src/CodeCoverage.php

```diff
@@ -388,20 +410,23 @@ public function merge(self $that): void
             // we should compare the lines if any of two contains data
             $compareLineNumbers = \array_unique(
                 \array_merge(
-                    \array_keys($this->data[$file]),
-                    \array_keys($that->data[$file])
+                    \array_keys($this->data[$file]['lines']),
+                    \array_keys($that->data[$file]['lines']) // can this be $fileData?
                 )
             );
 
             foreach ($compareLineNumbers as $line) {
-                $thatPriority = $this->getLinePriority($that->data[$file], $line);
-                $thisPriority = $this->getLinePriority($this->data[$file], $line);
+                $thatPriority = $this->getLinePriority($that->data[$file]['lines'], $line);
+                $thisPriority = $this->getLinePriority($this->data[$file]['lines'], $line);
 
                 if ($thatPriority > $thisPriority) {
-                    $this->data[$file][$line] = $that->data[$file][$line];
-                } elseif ($thatPriority === $thisPriority && \is_array($this->data[$file][$line])) {
-                    $this->data[$file][$line] = \array_unique(
-                        \array_merge($this->data[$file][$line], $that->data[$file][$line])
+                    $this->data[$file]['lines'][$line] = $that->data[$file]['lines'][$line];
+                } elseif ($thatPriority === $thisPriority && \is_array($this->data[$file]['lines'][$line])) {
+                    if ($line['pathCovered'] === true) {
+                        $this->data[$file]['lines']['line']['pathCovered'] = $line['pathCovered'];
```

Comment:
`'line'` should be `$line`, if I'm following this correctly.

Reply:
It should be an int. Also right - good catch.

Reply:
Changing the variable to `$lineNumber` should aid in readability.

Similarly, for other loops, `foreach ($fileData['lines'] as $lineNumber => $flag) {`